### ISSUE TITLE
Issue with starting sessions on first launch

### ISSUE THREAD

**Issue by arduano** [sddm/sddm#1800]
**TL;DR:** SDDM doesn't function properly until I `sudo systemctl restart sddm` each boot

I'm not sure how to word this properly but I'm convinced it's an SDDM issue, I'll try explaining what happened and the troubleshooting I did. Basically:

- I just did a full system update (Arch, KDE plasma/x11, SDDM), the issue started on the reboot right after I updated
- When trying to log in on the SDDM login screen on my main user acc, it just shows the plasma spinning gear for a few seconds until it just freezes and nothing loads. Other TTYs still show fine.
- Running startx loads plasma fine
- Logging in as my recovery user account through SDDM works fine
- At this point, I tried a bunch of KDE related troubleshooting, issue kept persisting between many reboots
- After a few reboots, I had the idea of trying to enable autologin to see what that changes. However, to my surprise, after enabling it, I would see a brief flash of the KDE wallet password request window, before the SDDM login screen takes over, even though I have auto login enabled. When I try to enter my password in this login screen, I get the same issue as above.
- I got very confused, and just tried running `sudo systemctl restart sddm`, somehow that worked and got me logged in.
- Rebooted, but the same as above happened again.
Here are the SDDM logs between boot and the SDDM login screen taking over with autologin enabled:

<details>
  <summary>SDDM logs on boot</summary>
<pre>
Sep 24 17:37:32 main-pc systemd[1]: Started Simple Desktop Display Manager.
Sep 24 17:37:32 main-pc sddm[1060]: Initializing...
Sep 24 17:37:32 main-pc sddm[1060]: Starting...
Sep 24 17:37:32 main-pc sddm[1060]: Logind interface found
Sep 24 17:37:32 main-pc sddm[1060]: Adding new display...
Sep 24 17:37:32 main-pc sddm[1060]: Loaded empty theme configuration
Sep 24 17:37:32 main-pc sddm[1060]: Xauthority path: "/run/sddm/xauth_WUQuIL"
Sep 24 17:37:32 main-pc sddm[1060]: Using VT 2
Sep 24 17:37:32 main-pc sddm[1060]: Display server starting...
Sep 24 17:37:32 main-pc sddm[1060]: Writing cookie to "/run/sddm/xauth_WUQuIL"
Sep 24 17:37:32 main-pc sddm[1060]: Running: /usr/bin/X -nolisten tcp -background none -seat seat0 vt2 -auth /run/sddm/xauth_WUQuIL -noreset -displayfd 16
Sep 24 17:37:34 main-pc sddm[1060]: Setting default cursor
Sep 24 17:37:34 main-pc sddm[1060]: Running display setup script  "/usr/share/sddm/scripts/Xsetup"
Sep 24 17:37:34 main-pc sddm[1060]: Display server started.
Sep 24 17:37:34 main-pc sddm[1060]: Reading from "/usr/local/share/xsessions/plasma.desktop"
Sep 24 17:37:34 main-pc sddm[1060]: Reading from "/usr/share/xsessions/plasma.desktop"
Sep 24 17:37:34 main-pc sddm[1060]: Session "/usr/share/xsessions/plasma.desktop" selected, command: "/usr/bin/startplasma-x11" for VT 2
Sep 24 17:37:34 main-pc sddm-helper[1117]: [PAM] Starting...
Sep 24 17:37:34 main-pc sddm-helper[1117]: [PAM] Authenticating...
Sep 24 17:37:34 main-pc sddm-helper[1117]: pam_kwallet5(sddm-autologin:auth): pam_kwallet5: pam_sm_authenticate
Sep 24 17:37:34 main-pc sddm-helper[1117]: [PAM] Preparing to converse...
Sep 24 17:37:34 main-pc sddm-helper[1117]: pam_kwallet5(sddm-autologin:auth): pam_kwallet5: Couldn't get password (it is empty)
Sep 24 17:37:34 main-pc sddm-helper[1117]: [PAM] Conversation with 1 messages
Sep 24 17:37:34 main-pc sddm-helper[1117]: pam_kwallet5(sddm-autologin:auth): pam_kwallet5: Empty or missing password, doing nothing
Sep 24 17:37:34 main-pc sddm-helper[1117]: [PAM] returning.
Sep 24 17:37:34 main-pc sddm[1060]: Authentication for user  "arduano"  successful
Sep 24 17:37:34 main-pc sddm-helper[1117]: pam_kwallet5(sddm-autologin:setcred): pam_kwallet5: pam_sm_setcred
Sep 24 17:37:34 main-pc sddm-helper[1117]: pam_unix(sddm-autologin:session): session opened for user arduano(uid=1000) by arduano(uid=0)
Sep 24 17:37:35 main-pc sddm-helper[1117]: pam_kwallet5(sddm-autologin:session): pam_kwallet5: pam_sm_open_session
Sep 24 17:37:35 main-pc sddm-helper[1117]: pam_kwallet5(sddm-autologin:session): pam_kwallet5: open_session called without kwallet5_key
Sep 24 17:37:35 main-pc sddm-helper[1117]: Writing cookie to "/tmp/xauth_BCvIfq"
Sep 24 17:37:35 main-pc sddm-helper[1117]: Starting X11 session: "" "/usr/share/sddm/scripts/Xsession \"/usr/bin/startplasma-x11\""
Sep 24 17:37:35 main-pc sddm[1060]: Session started true
Sep 24 17:37:37 main-pc sddm[1060]: Auth: sddm-helper exited successfully
Sep 24 17:37:37 main-pc sddm[1060]: Display server stopping...
Sep 24 17:37:37 main-pc sddm[1060]: Display server stopped.
Sep 24 17:37:37 main-pc sddm[1060]: Running display stop script  ("/usr/share/sddm/scripts/Xstop")
Sep 24 17:37:37 main-pc sddm[1060]: Removing display SDDM::Display(0x5570cd673920) ...
Sep 24 17:37:37 main-pc sddm[1060]: Adding new display...
Sep 24 17:37:37 main-pc sddm[1060]: Loaded empty theme configuration
Sep 24 17:37:37 main-pc sddm[1060]: Xauthority path: "/run/sddm/xauth_CpKymE"
Sep 24 17:37:37 main-pc sddm[1060]: Using VT 1
Sep 24 17:37:37 main-pc sddm[1060]: Display server starting...
Sep 24 17:37:37 main-pc sddm[1060]: Writing cookie to "/run/sddm/xauth_CpKymE"
Sep 24 17:37:37 main-pc sddm[1060]: Running: /usr/bin/X -nolisten tcp -background none -seat seat0 vt1 -auth /run/sddm/xauth_CpKymE -noreset -displayfd 17
Sep 24 17:37:39 main-pc sddm[1060]: Setting default cursor
Sep 24 17:37:39 main-pc sddm-helper[2241]: pam_unix(sddm-greeter:session): session opened for user sddm(uid=967) by (uid=0)
Sep 24 17:37:39 main-pc sddm[1060]: Running display setup script  "/usr/share/sddm/scripts/Xsetup"
Sep 24 17:37:39 main-pc sddm[1060]: Display server started.
Sep 24 17:37:39 main-pc sddm[1060]: Socket server starting...
Sep 24 17:37:39 main-pc sddm[1060]: Socket server started.
Sep 24 17:37:39 main-pc sddm[1060]: Loading theme configuration from "/usr/share/sddm/themes/breeze/theme.conf"
Sep 24 17:37:39 main-pc sddm[1060]: Greeter starting...
Sep 24 17:37:39 main-pc sddm-helper[2241]: [PAM] Starting...
Sep 24 17:37:39 main-pc sddm-helper[2241]: [PAM] Authenticating...
Sep 24 17:37:39 main-pc sddm-helper[2241]: [PAM] returning.
Sep 24 17:37:39 main-pc sddm-helper[2241]: Writing cookie to "/tmp/xauth_uuiaiP"
Sep 24 17:37:39 main-pc sddm-helper[2241]: Starting X11 session: "" "/usr/bin/sddm-greeter --socket /tmp/sddm-:0-XekWaS --theme /usr/share/sddm/themes/breeze"
Sep 24 17:37:39 main-pc sddm[1060]: Greeter session started successfully
Sep 24 17:37:39 main-pc sddm[1060]: Message received from greeter: Connect
</pre>
</details>

After grabbing these logs, I started writing this gh issue. But just to make sure, I switched off autologin to see if restarting sddm solves the previous issue I had, but seems like it was fine somehow.

In terms of having a usable computer, seems like my first problem fixed itself somehow, but this was fairly strange behavior (especially the autologin one) so I'm opening this issue anyway. I'm not sure if there's much else I can reproduce, but let me know if I should try providing any more specific details.

**Comment by Vogtinator** [comment]
Yeah, autologin is sometimes broken for a so far unknown reason, probably a race condition.

**Comment by arduano** [comment]
Ok SDDM not loading after the login screen \*without* autologin happened again, here's the logs:

<details>
<summary> Logs until the spinning gear stops spinning and the PC is just stuck </summary>
<pre>
Oct 05 02:32:20 main-pc systemd[1]: Started Simple Desktop Display Manager.
Oct 05 02:32:20 main-pc sddm[1056]: Initializing...
Oct 05 02:32:20 main-pc sddm[1056]: Starting...
Oct 05 02:32:20 main-pc sddm[1056]: Logind interface found
Oct 05 02:32:20 main-pc sddm[1056]: Adding new display...
Oct 05 02:32:20 main-pc sddm[1056]: Loaded empty theme configuration
Oct 05 02:32:20 main-pc sddm[1056]: Xauthority path: "/run/sddm/xauth_hsAIIn"
Oct 05 02:32:20 main-pc sddm[1056]: Using VT 2
Oct 05 02:32:20 main-pc sddm[1056]: Display server starting...
Oct 05 02:32:20 main-pc sddm[1056]: Writing cookie to "/run/sddm/xauth_hsAIIn"
Oct 05 02:32:20 main-pc sddm[1056]: Running: /usr/bin/X -nolisten tcp -background none -seat seat0 vt2 -auth /run/sddm/xauth_hsAIIn -noreset -displayfd 16
Oct 05 02:32:23 main-pc sddm[1056]: Setting default cursor
Oct 05 02:32:24 main-pc sddm[1056]: Could not setup default cursor
Oct 05 02:32:24 main-pc sddm[1056]: Running display setup script  "/usr/share/sddm/scripts/Xsetup"
Oct 05 02:32:24 main-pc sddm[1056]: Display server started.
Oct 05 02:32:24 main-pc sddm[1056]: Socket server starting...
Oct 05 02:32:24 main-pc sddm[1056]: Socket server started.
Oct 05 02:32:24 main-pc sddm[1056]: Loading theme configuration from "/usr/share/sddm/themes/breeze/theme.conf"
Oct 05 02:32:24 main-pc sddm[1056]: Greeter starting...
Oct 05 02:32:24 main-pc sddm-helper[1707]: [PAM] Starting...
Oct 05 02:32:24 main-pc sddm-helper[1707]: [PAM] Authenticating...
Oct 05 02:32:24 main-pc sddm-helper[1707]: [PAM] returning.
Oct 05 02:32:24 main-pc sddm-helper[1707]: pam_unix(sddm-greeter:session): session opened for user sddm(uid=967) by (uid=0)
Oct 05 02:32:24 main-pc sddm[1056]: Display server stopped.
Oct 05 02:32:24 main-pc sddm[1056]: Running display stop script  ("/usr/share/sddm/scripts/Xstop")
Oct 05 02:32:24 main-pc sddm[1056]: Socket server stopping...
Oct 05 02:32:24 main-pc sddm[1056]: Socket server stopped.
Oct 05 02:32:24 main-pc sddm[1056]: Removing display SDDM::Display(0x561dc4272960) ...
Oct 05 02:32:24 main-pc sddm[1056]: Adding new display...
Oct 05 02:32:24 main-pc sddm[1056]: Loaded empty theme configuration
Oct 05 02:32:24 main-pc sddm[1056]: Xauthority path: "/run/sddm/xauth_zOLwql"
Oct 05 02:32:24 main-pc sddm[1056]: Using VT 1
Oct 05 02:32:24 main-pc sddm[1056]: Display server starting...
Oct 05 02:32:24 main-pc sddm[1056]: Writing cookie to "/run/sddm/xauth_zOLwql"
Oct 05 02:32:24 main-pc sddm[1056]: Running: /usr/bin/X -nolisten tcp -background none -seat seat0 vt1 -auth /run/sddm/xauth_zOLwql -noreset -displayfd 19
Oct 05 02:32:25 main-pc sddm-helper[1707]: Writing cookie to "/tmp/xauth_JrsZoc"
Oct 05 02:32:25 main-pc sddm-helper[1707]: Starting X11 session: "" "/usr/bin/sddm-greeter --socket /tmp/sddm-:0-vcgvAg --theme /usr/share/sddm/themes/breeze"
Oct 05 02:32:27 main-pc sddm[1056]: Setting default cursor
Oct 05 02:32:27 main-pc sddm[1056]: Running display setup script  "/usr/share/sddm/scripts/Xsetup"
Oct 05 02:32:27 main-pc sddm[1056]: Display server started.
Oct 05 02:32:27 main-pc sddm[1056]: Socket server starting...
Oct 05 02:32:27 main-pc sddm[1056]: Socket server started.
Oct 05 02:32:27 main-pc sddm[1056]: Loading theme configuration from "/usr/share/sddm/themes/breeze/theme.conf"
Oct 05 02:32:27 main-pc sddm[1056]: Greeter starting...
Oct 05 02:32:27 main-pc sddm-helper[1738]: [PAM] Starting...
Oct 05 02:32:27 main-pc sddm-helper[1738]: [PAM] Authenticating...
Oct 05 02:32:27 main-pc sddm-helper[1738]: [PAM] returning.
Oct 05 02:32:32 main-pc sddm[1056]: QProcess: Destroyed while process ("/usr/lib/sddm/sddm-helper") is still running.
Oct 05 02:32:32 main-pc sddm-helper[1738]: pam_unix(sddm-greeter:session): session opened for user sddm(uid=967) by (uid=0)
Oct 05 02:32:32 main-pc sddm-helper[1738]: Writing cookie to "/tmp/xauth_dtIDyu"
Oct 05 02:32:32 main-pc sddm-helper[1738]: Starting X11 session: "" "/usr/bin/sddm-greeter --socket /tmp/sddm-:0-OaHmGy --theme /usr/share/sddm/themes/breeze"
Oct 05 02:32:32 main-pc sddm[1056]: Greeter session started successfully
Oct 05 02:32:32 main-pc sddm[1056]: Message received from greeter: Connect
Oct 05 02:32:43 main-pc sddm[1056]: Message received from greeter: Login
Oct 05 02:32:43 main-pc sddm[1056]: Reading from "/usr/share/xsessions/plasma.desktop"
Oct 05 02:32:43 main-pc sddm[1056]: Session "/usr/share/xsessions/plasma.desktop" selected, command: "/usr/bin/startplasma-x11" for VT 1
Oct 05 02:32:44 main-pc sddm-helper[1797]: [PAM] Starting...
Oct 05 02:32:44 main-pc sddm-helper[1797]: [PAM] Authenticating...
Oct 05 02:32:44 main-pc sddm-helper[1797]: [PAM] Preparing to converse...
Oct 05 02:32:44 main-pc sddm-helper[1797]: [PAM] Conversation with 1 messages
Oct 05 02:32:44 main-pc sddm-helper[1797]: pam_kwallet5(sddm:auth): pam_kwallet5: pam_sm_authenticate
Oct 05 02:32:44 main-pc sddm-helper[1797]: [PAM] returning.
Oct 05 02:32:44 main-pc sddm[1056]: Authentication for user  "arduano"  successful
Oct 05 02:32:44 main-pc sddm-helper[1797]: pam_kwallet5(sddm:setcred): pam_kwallet5: pam_sm_setcred
Oct 05 02:32:44 main-pc sddm-helper[1797]: pam_unix(sddm:session): session opened for user arduano(uid=1000) by arduano(uid=0)
Oct 05 02:32:44 main-pc sddm[1056]: Auth: sddm-helper exited successfully
Oct 05 02:32:44 main-pc sddm[1056]: Greeter stopped. SDDM::Auth::HELPER_SUCCESS
Oct 05 02:32:44 main-pc sddm-helper[1797]: pam_kwallet5(sddm:session): pam_kwallet5: pam_sm_open_session
Oct 05 02:32:44 main-pc sddm-helper[1797]: Writing cookie to "/tmp/xauth_eFLVXX"
Oct 05 02:32:44 main-pc sddm-helper[1797]: Starting X11 session: "" "/usr/share/sddm/scripts/Xsession \"/usr/bin/startplasma-x11\""
Oct 05 02:32:44 main-pc sddm[1056]: Session started true
</pre>
</details>

Can confirm that `sudo systemctl restart sddm` fixes this issue too.

**Comment by arduano** [comment]
It happened again on the subsequent restart, though I had to run restart twice before it let me in. Thankfully I can easily SSH from my phone, though this is quite annoying.

Please let me know if there's any logs or diagnostics I could provide to help

**Comment by githubkusi** [comment]
same behavior here, black screen on autologin into a wayland session, `sudo systemctl restart sddm ` fixes it
